packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -329,6 +410,77 @@ export async function generatePersistedQueryManifest(
     });
   }
 
+  return sources;
+}
+
+// Unfortunately globby does not guarantee deterministic file sorting so we
+// apply some sorting on the files in this function.
+//
+// https://github.com/sindresorhus/globby/issues/131
+/** @internal */
+export async function getFilepaths(
+  documents: string | string[] | CustomDocumentSourceConfig,
+) {
+  if (isCustomDocumentsSource(documents)) {
+    const sources = documents[CUSTOM_DOCUMENTS_SOURCE]();
+
+    return [
+      ...new Set(
+        sources.filter(({ file }) => file.path).map(({ file }) => file.path!),
+      ),
+    ];
+  }
+
+  return [...uniq(await globby(documents))].sort((a, b) => a.localeCompare(b));
+}
+
+/** @internal */
+export async function generatePersistedQueryManifest(
+  config: PersistedQueryManifestConfig = {},
+  configFilePath: string | undefined,
+): Promise<PersistedQueryManifest> {
+  const {
+    documents = defaults.documents,
+    createOperationId = defaults.createOperationId,
+  } = config;
+
+  const configFile = vfile({
+    path: configFilePath
+      ? relative(process.cwd(), configFilePath)
+      : "<virtual>",
+  });
+
+  const sources = isCustomDocumentsSource(documents)
+    ? documents[CUSTOM_DOCUMENTS_SOURCE]()
+    : await fromFilepathList(documents);
+
+  const operationsByName = new Map<string, DocumentSource[]>();
+
+  for (const source of sources) {
+    if (!source.node) {
+      continue;
+    }
+
+    // We delegate validation to the functions that return the document sources.
+    // We just need to record the operations here to sort them in the manifest
+    // output.
+    visit(source.node, {
+      OperationDefinition(node) {
+        const name = node.name?.value;
+
+        if (!name) {
+          return false;
+        }
+
+        const sources = operationsByName.get(name) ?? [];
+
+        operationsByName.set(name, [...sources, source]);
+
+        return false;
+      },
+    });
+  }
+
   maybeReportErrorsAndExit(uniq(sources.map((source) => source.file)));
 
   // Using createFragmentRegistry means our minimum AC version is 3.7. We can
```

Comment:
I would prefer that we leave out the fragment registry if we've parsed our documents from a format that's supposed to have already done the fragment stuff.

Reply:
Done! [8243f09](https://github.com/apollographql/apollo-utils/pull/414/commits/8243f09a400f4333832b1d8ee1e56d23ae62f1ec)